hal/api/toolchain.h

```diff
@@ -225,6 +225,35 @@
 #endif
 #endif
 
+/* Define '__RAMFUNC' as an attribute to mark a function as residing in RAM.
+ * Use of __RAMFUNC puts a function in the .data section--i.e. the
+ * initialized data section. This will be copied into RAM automatically by the
+ * startup sequence. */
+#ifndef __RAMFUNC
+#if defined(__GNUC__) || defined(__clang__) // GCC and llvm/clang
+#define __RAMFUNC __attribute__ ((section (".data#"), noinline)) /* The '#' following ".data" needs a bit of
```

Comment:
Have you exhausted the possibilities for a proper fix?

I recall from the last PR that ".data.ramfunc" complains because it expects ".data.*" sections to be non-executable. 

Have you tried a completely new section name, like ".ramfuncs"? Would mean adding it to the linker map.